pkg/workceptor/remote_work.go

```diff
@@ -68,17 +75,34 @@ func (rw *remoteUnit) connectToRemote(ctx context.Context) (net.Conn, *bufio.Rea
 }
 
 // getConnection retries connectToRemote until connected or the context expires
-func (rw *remoteUnit) getConnection(ctx context.Context) (net.Conn, *bufio.Reader) {
+func (rw *remoteUnit) getConnection(mw *utils.JobContext) (net.Conn, *bufio.Reader) {
 	connectDelay := utils.NewIncrementalDuration(SuccessWorkSleep, MaxWorkSleep, 1.5)
 	for {
-		conn, reader, err := rw.connectToRemote(ctx)
+		conn, reader, err := rw.connectToRemote(mw)
 		if err == nil {
 			return conn, reader
 		}
-		logger.Debug("Connection to %s failed with error: %s",
-			rw.Status().ExtraData.(*remoteExtraData).RemoteNode, err)
+		status := rw.Status()
+		logger.Warning("Connection to %s failed with error: %s",
+			status.ExtraData.(*remoteExtraData).RemoteNode, err)
+		errStr := err.Error()
+		if strings.Contains(errStr, "CRYPTO_ERROR") {
+			detailMsg := "Incorrect tlsclient to remote service"
+			if status.Detail != detailMsg {
+				rw.UpdateFullStatus(func(status *StatusFileData) {
+					status.Detail = detailMsg
+				})
+			}
+			if status.ExtraData.(*remoteExtraData).RemoteStarted == false {
+				rw.UpdateFullStatus(func(status *StatusFileData) {
+					status.State = WorkStateFailed
+				})
+				mw.WorkerDone()
```

Comment:
not calling `WorkDone()` here causes subsequent `release` or `cancel` commands on this work unit to hang.

Reply:
I think the problem here is that you ought to `return` - as things stand you're calling `WorkerDone` but then continuing around the function, which is blocking.  If you _did_ return then I think the outer function would take care of calling `WorkerDone` for you.  It's also super important to make sure that `WorkerDone` only gets called _once_ per dispatched worker goroutine, which I think won't be the case here (since, whenever this function _does_ manage to exit the loop, it will return to its caller, who has a deferred WorkerDone ready to go).

Reply:
this is fixed up now

now all WorkerDone() calls for the job are in a single method, `runAndMonitor`